src/structuring_element.jl

```diff
@@ -0,0 +1,356 @@
+abstract type MorphologySE{N} end
+
+"""
+    SEMask{N}()
+
+The N-dimensional structuring element in terms of bool array as mask. Typically, `true`
+represents the foreground and `false` represents the background.
+"""
+struct SEMask{N} <: MorphologySE{N} end
+
+"""
+    SEOffset{N}()
+
+The N-dimensional structuring element in terms of displacement offset.
+"""
+struct SEOffset{N} <: MorphologySE{N} end
+
+"""
+    SEDiamond{N}([size], [dims]; r=1)
+
+The N-dimensional diamond shape structuring element.
```

Comment:
What is `dims` for? Is it a list of dimensions over which this operates? How does that differ from having `size[d] = 0`? Why is `r=1` the default rather than `r[d] = size[d] ÷ 2`? For anisotropic imaging (commonplace in 3d biomedical imaging), presumably you might want to look several pixels in the high-resolution axes but only at nearest-neighbors in the low-resolution axes.

Reply:
For the `r::Int` decision, this is mainly because I'm lazy... because if we extend `r` to be a tuple, implementing an efficient algorithm to check if a given point is within the N-dimensional diamond(polytope) isn't something trivial.

Reply:
Separating out `dims` allows me to generalize the definition a bit to skip certain dimensions as `extremefilt!` allows, yet without the need to play with the is-point-in-polytope algorithm.